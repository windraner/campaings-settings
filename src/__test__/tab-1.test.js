import React from 'react';
import Tab1 from '../components/main-content-section/tab-block/tab-1/tab-1';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
  isShoppableLink: true,
  name: 'Summer Sales 10%',
  productCatalogue: 'Product catalogue2',
  productsCatalogueList: [
    'Product catalogue1',
    'Product catalogue2',
    'Product catalogue3',
    'Product catalogue4',
  ],
  stateHandler() {},
  switchHandler() {},
};

describe('<Tab1 />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Tab1 {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
