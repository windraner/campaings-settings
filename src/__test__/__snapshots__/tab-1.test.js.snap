// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab1 /> matches the snapshot 1`] = `
<Fragment>
  <TabTitle
    text="Campaign details"
  />
  <TabContainer>
    <TextInput
      field="name"
      required={true}
      stateHandler={[Function]}
      title="Name"
      value="Summer Sales 10%"
    />
    <div
      className="campaigns-settings__divider-mt-32"
    />
    <TextareaInput
      className="campaigns-settings__min-h-120"
      field="description"
      stateHandler={[Function]}
      title="Description"
      value="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
    />
    <div
      className="campaigns-settings__divider-mt-32"
    />
    <Dropdown
      field="productCatalogue"
      productsCatalogueList={
        Array [
          "Product catalogue1",
          "Product catalogue2",
          "Product catalogue3",
          "Product catalogue4",
        ]
      }
      stateHandler={[Function]}
      title="Link campaign to a product catalogue?"
      value="Product catalogue2"
    />
    <hr
      className="campaigns-settings__border-separator"
    />
    <Switch
      field="isShoppableLink"
      isChecked={true}
      switchHandler={[Function]}
      text="Enable shoppable link editing"
    />
  </TabContainer>
</Fragment>
`;
